README.md

```diff
@@ -1,14 +1,110 @@
-# Library of cryptographic functions for Dart
+# Cryptographic digest functions for Dart
 
-A set of cryptographic functions implemented in pure Dart.
+A set of cryptographic digest functions implemented in pure
+Dart. Cryptographic digest is also known as cryptographic hashing.
+
+The following digest/hashing algorithms are supported:
 
 * MD5
-* SHA1
-* SHA265
-* HMAC
-* Converting a list of bytes into a hex string
-* Converting a list of bytes into a base64-encoded string.
-* Converting a Base 64 encoded String into list of bytes.
+* SHA-1
+* SHA-256
+* HMAC (i.e. HMAC-MD5, HMAC-SHA1, HMAC-SHA256)
+
+## Usage
+
+### Digest on a single input
+
+Use the `sha256` property to obtain an instance of the `Sha256` class.
+Or use the `sha1` or `md5` properties to use those other digest
+algorithms.
+
+Invoke the `convert` method, passing the input data as a sequence of
+bytes (i.e. `List<int>`). It returns a `Digest` object, where the
+digest value can be obtained as bytes in a `List<int>` or encoded in
+hexadecimal as a `String`.
+
+```dart
+import 'package:crypto/crypto.dart';
+import 'dart:convert'; // for the UTF8.encode method
+
+  ...
+
+  var dataString = "foobar"; // input data being hashed
+
+  var bytes = UTF8.encode(dataString);
+
+  var digest = sha1.convert(bytes);
+
+  print("Digest as bytes: ${digest.bytes}");
+  print("Digest as a hexadecimal string: $digest");
+```
+
+In the above example, the data being digested is a `String`, so the
+`UTF8.encode` method was used to convert it into a sequence of bytes.
+
+**Warning:** The input data represents a sequence of bytes, so the
+`List<int>` **must not** contain integers outside the range 0-255.
+Do not use the `codeUnits` property of a `String`, unless it is
+_guaranteed_ the string will never contain characters outside the
+range of 0-255. Even then, characters in the range of 128-255 might
+produce the wrong result.
```

Comment:
I don't think this is necessary here—it's not really a property of this library, but of the general way Dart handles binary data and strings.